### PR TITLE
Run ./hack/release.sh from the release branch

### LINKED ISSUE #1262: dot-release is running the ./hack/release.sh in master branch
`eventing-contrib` repo is doing a dot-release but the job ended up with an error.

From the job logs https://prow.knative.dev/log?job=ci-knative-eventing-contrib-dot-release&id=1161353664162435072, we can see the error is `2019/08/13 19:14:03 Error enumerating files: lstat github/config/: no such file or directory`. However, `github/config/` is a path written in the `./hack/release.sh` of master branch but not 0.8 branch.

Our dot-release job should use `./hack/release.sh` in branch 0.8, instead of master.

/cc @adrcunha 

FYI @chaodaiG since he found the reason of this bug.

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
Make sure `./hack/release.sh` is finally run from the release branch.

**Which issue(s) this PR fixes**:
Fixes #1262 

/cc @chaodaiG 
FYI @grantr @mattmoor

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -509,6 +509,7 @@ function main() {
     # HACK HACK HACK
     # Rerun the release script from the release branch. Fixes https://github.com/knative/test-infra/issues/1262
     ./hack/release.sh "$@"
+    exit "$?"
```
Comment (chaodaiG) [inline scripts/release.sh]: are you sure stop here?

Comment (chizhg) [reply scripts/release.sh]: Yes, at this point, the logic below would have been executed in the `./hack/release.sh` in the release branch, so we should stop here.

File: scripts/release.sh [hunk 2]
```diff
@@ -498,9 +498,23 @@ function publish_artifacts() {
 
 # Entry point for a release script.
 function main() {
+  parse_flags "$@"
+
+  # Checkout specific branch, if necessary
+  local current_branch="$(git rev-parse --abbrev-ref HEAD)"
```
Comment (chaodaiG) [inline scripts/release.sh]: I know it's rare, but `local` swallow the error after `=`
```suggestion
  local current_branch
  current_branch="$(git rev-parse --abbrev-ref HEAD)"
```

**Comment by chaodaiG** [comment]
/lgtm

Feel free ignore my suggestion if it's an urgent fix

**Comment by chizhg** [comment]
> /lgtm
> 
> Feel free ignore my suggestion if it's an urgent fix

It's not. I need to do some more test before merging it in.

**Comment by chizhg** [comment]
I've done some tests and it works.
/hold cancel
/assign @chaodaiG

**Pull request merged**